ansible_runner/runner_config.py

```diff
@@ -563,26 +606,136 @@ def wrap_args_for_containerization(self, args):
         new_args = [self.process_isolation_executable]
         new_args.extend(['run', '--rm', '--tty', '--interactive'])
         new_args.extend(["--workdir", "/runner/project"])
+        new_args.extend(["-e", "LAUNCHED_BY_RUNNER=1"])
 
         def _ensure_path_safe_to_mount(path):
             if path in ('/home', '/usr'):
                 raise ConfigurationError("When using containerized execution, cannot mount /home or /usr")
 
-        # map private data to /runner
-        # .. using Z to modify the selinux label of the _host_ file / directory
-        # (see https://docs.docker.com/storage/bind-mounts/#configure-the-selinux-label
-        #  for usage and potential side-effects)
         _ensure_path_safe_to_mount(self.private_data_dir)
-        new_args.extend(["-v", "{}:/runner:Z".format(self.private_data_dir)])
+
+        if not self.cli_execenv_cmd:
```

Comment:
I'm particularly worried about the overall state machine for command line generation, the complexity is a bit out of control and I'd like to change this to a more composable system.